### PR TITLE
Refactor and consolidate time window logic

### PR DESCRIPTION
**Pull request by thebentern**

### CODE REVIEW

File: src/graphics/EInkDynamicDisplay.cpp [hunk 1]
```diff
@@ -239,7 +240,7 @@ void EInkDynamicDisplay::checkRateLimiting()
```
Comment (fifieldt) [inline src/graphics/EInkDynamicDisplay.cpp:240]: Can probably remove the now variable compare directly with millis above?

File: src/input/ExpressLRSFiveWay.cpp [hunk 2]
```diff
@@ -88,7 +89,7 @@ void ExpressLRSFiveWay::update(int *keyValue, bool *keyLongPressed)
         if (newKey == NO_PRESS) {
```
Comment (fifieldt) [inline src/input/ExpressLRSFiveWay.cpp:89]: Can probably remove the now var and set keyDownStart = millis()

File: src/input/ExpressLRSFiveWay.cpp [hunk 3]
```diff
@@ -101,7 +102,7 @@ void ExpressLRSFiveWay::update(int *keyValue, bool *keyLongPressed)
         }
         // else still pressing, waiting for long if not already signaled
         else if (!isLongPressed) {
-            if ((now - keyDownStart) > KEY_LONG_PRESS_MS) {
+            if (!Throttle::isWithinTimespanMs(keyDownStart, KEY_LONG_PRESS_MS)) {
                 *keyValue = keyInProcess;
                 *keyLongPressed = true;
                 isLongPressed = true;
```
Comment (fifieldt) [inline src/input/ExpressLRSFiveWay.cpp:108]: There are some calls below in nt32_t ExpressLRSFiveWay::runOnce() that might be worth looking into too...

File: src/mesh/LR11x0Interface.cpp [hunk 4]
```diff
@@ -278,12 +280,13 @@ template <typename T> bool LR11x0Interface<T>::isActivelyReceiving()
         uint32_t now = millis();
         if (!activeReceiveStart) {
             activeReceiveStart = now;
```
Comment (fifieldt) [inline src/mesh/LR11x0Interface.cpp:282]: activeReceiveStart = millis() ?

File: src/mesh/PacketHistory.cpp [hunk 5]
```diff
@@ -32,8 +33,9 @@ bool PacketHistory::wasSeenRecently(const meshtastic_MeshPacket *p, bool withUpd
     auto found = recentPackets.find(r);
```
Comment (fifieldt) [inline src/mesh/PacketHistory.cpp:33]: r.rxTimeMsec = millis()

File: src/mesh/PacketHistory.cpp [hunk 6]
```diff
@@ -69,7 +71,7 @@ void PacketHistory::clearExpiredRecentPackets()
     LOG_DEBUG("recentPackets size=%ld\n", recentPackets.size());
```
Comment (fifieldt) [inline src/mesh/PacketHistory.cpp:71]: now var no longer needed

File: src/mesh/SX126xInterface.cpp [hunk 7]
```diff
@@ -322,12 +324,13 @@ template <typename T> bool SX126xInterface<T>::isActivelyReceiving()
         uint32_t now = millis();
         if (!activeReceiveStart) {
             activeReceiveStart = now;
```
Comment (fifieldt) [inline src/mesh/SX126xInterface.cpp:326]: activeReceiveStart = millis();

File: src/mesh/SX128xInterface.cpp [hunk 8]
```diff
@@ -297,12 +298,13 @@ template <typename T> bool SX128xInterface<T>::isActivelyReceiving()
         uint32_t now = millis();
         if (!activeReceiveStart) {
             activeReceiveStart = now;
```
Comment (fifieldt) [inline src/mesh/SX128xInterface.cpp:300]: activeReceiveStart = now;

Comment (fifieldt) [reply src/mesh/SX128xInterface.cpp:300]: mills()*

File: src/mesh/StreamAPI.cpp [hunk 9]
```diff
@@ -23,7 +24,7 @@ int32_t StreamAPI::readStream()
     uint32_t now = millis();
```
Comment (fifieldt) [inline src/mesh/StreamAPI.cpp:24]: now var no longer needed

File: src/modules/NeighborInfoModule.cpp [hunk 10]
```diff
@@ -87,7 +88,8 @@ void NeighborInfoModule::cleanUpNeighbors()
     NodeNum my_node_id = nodeDB->getNodeNum();
     for (auto it = neighbors.rbegin(); it != neighbors.rend();) {
         // We will remove a neighbor if we haven't heard from them in twice the broadcast interval
-        if ((now - it->last_rx_time > it->node_broadcast_interval_secs * 2) && (it->node_id != my_node_id)) {
```
Comment (fifieldt) [inline src/modules/NeighborInfoModule.cpp:90]: check that getTime() is equivalent to millis()

Comment (fifieldt) [reply src/modules/NeighborInfoModule.cpp:90]: + remove now variable.

File: src/modules/NodeInfoModule.cpp [hunk 11]
```diff
@@ -69,11 +70,11 @@ meshtastic_MeshPacket *NodeInfoModule::allocReply()
     }
     uint32_t now = millis();
```
Comment (fifieldt) [inline src/modules/NodeInfoModule.cpp:71]: set         lastSentToMesh = millis() below and save a variable!

File: src/modules/PowerStressModule.cpp [hunk 12]
```diff
@@ -114,7 +115,7 @@ int32_t PowerStressModule::runOnce()
                 break;
             case meshtastic_PowerStressMessage_Opcode_CPU_FULLON: {
                 uint32_t start_msec = millis();
```
Comment (fifieldt) [inline src/modules/PowerStressModule.cpp:117]: start_msec is set to millis() ... this looks strange

File: src/modules/RemoteHardwareModule.cpp [hunk 13]
```diff
@@ -120,7 +121,7 @@ int32_t RemoteHardwareModule::runOnce()
     if (moduleConfig.remote_hardware.enabled && watchGpios) {
```
Comment (fifieldt) [inline src/modules/RemoteHardwareModule.cpp:121]: Set  lastWatchMsec = millis(); and save a variable!

File: src/modules/SerialModule.cpp [hunk 14]
```diff
@@ -98,7 +99,7 @@ SerialModuleRadio::SerialModuleRadio() : MeshModule("SerialModuleRadio")
 bool SerialModule::checkIsConnected()
 {
     uint32_t now = millis();
```
Comment (fifieldt) [inline src/modules/SerialModule.cpp:101]: now variable no longer needed

File: src/modules/Telemetry/AirQualityTelemetry.cpp [hunk 15]
```diff
@@ -61,12 +62,12 @@ int32_t AirQualityTelemetryModule::runOnce()
...
-            airTime->isTxAllowedChannelUtil(config.device.role != meshtastic_Config_DeviceConfig_Role_SENSOR) &&
-            airTime->isTxAllowedAirUtil()) {
+        if ((lastSentToMesh == 0) ||
+            (!Throttle::isWithinTimespanMs(lastSentToMesh, Default::getConfiguredOrDefaultMsScaled(
+                                                               moduleConfig.telemetry.air_quality_interval,
+                                                               default_telemetry_broadcast_interval_secs, numOnlineNodes))) &&
+                airTime->isTxAllowedChannelUtil(config.device.role != meshtastic_Config_DeviceConfig_Role_SENSOR) &&
+                airTime->isTxAllowedAirUtil()) {
             sendTelemetry();
             lastSentToMesh = now;
```
Comment (fifieldt) [inline src/modules/Telemetry/AirQualityTelemetry.cpp:72]: lastSentToMesh =  millis() and delete now variable above.

File: src/modules/Telemetry/EnvironmentTelemetry.cpp [hunk 16]
```diff
@@ -156,15 +157,15 @@ int32_t EnvironmentTelemetryModule::runOnce()
         }
 
         uint32_t now = millis();
```
Comment (fifieldt) [inline src/modules/Telemetry/EnvironmentTelemetry.cpp:159]: now variable unneeded.

**Review by fifieldt — COMMENTED** [review]
you're going to love all the memory we can save!!!!

**Comment by fifieldt** [comment]
This is a good change. I can sense there's a future vision here ...

File: src/mesh/RadioLibInterface.cpp [hunk 17]
```diff
@@ -41,7 +42,7 @@ void LockingArduinoHal::spiTransfer(uint8_t *out, size_t len, uint8_t *in)
 
             uint32_t start = millis();
             while (digitalRead(busy)) {
-                if (millis() - start >= 2000) {
+                if (!Throttle::isWithinTimespanMs(start, 2000) {
```
Comment (fifieldt) [inline src/mesh/RadioLibInterface.cpp:45]: missing a brace at the end

File: src/modules/NeighborInfoModule.cpp (hunk 10 above)
Comment (fifieldt) [reply src/modules/NeighborInfoModule.cpp:90]: getTime:" * Returns the current time in seconds since the Unix epoch (January 1, 1970)."

**Comment by fifieldt** [comment]
@thebentern , I pushed a commit that fixed the syntax error and removed some of the unused variables. A few comments in the previous patch you might still like to take a look at.

File: src/mesh/wifi/WiFiAPClient.cpp [hunk 18]
```diff
@@ -142,7 +142,8 @@ static int32_t reconnectWiFi()
     }
 
 #ifndef DISABLE_NTP
-    if (WiFi.isConnected() && (((millis() - lastrun_ntp) > 43200000) || (lastrun_ntp == 0))) { // every 12 hours
+#include "Throttle.h"
```
Comment (fifieldt) [inline src/mesh/wifi/WiFiAPClient.cpp:145]: Including Throttle here causes build errors.

File: src/modules/PowerStressModule.cpp (hunk 12 above)
Comment (thebentern) [reply src/modules/PowerStressModule.cpp:117]: I thought so too. I think the intent here is to hamster wheel the CPU perhaps

**Review by fifieldt — APPROVED** [review]
Good to go

**Comment by gitbisector** [comment]
@mverch67 There are a number of places where 'seconds' are passed in rather than 'milliseconds' (NeighborInfo is one) in this commit.

**Comment by thebentern** [comment]
@gitbisector would you mind creating an issue bug?

**Pull request merged**